engine/test/components/test_cortex_upd_cmd.cc

```diff
@@ -0,0 +1,56 @@
+#include "commands/cortex_upd_cmd.h"
+#include "gtest/gtest.h"
+
+namespace {
+constexpr const auto kNewReleaseFolder = "./cortex-release";
+constexpr const auto kNewReleaseFile = "./cortex-release/cortexexe";
+constexpr const auto kCurReleaseFile = "./cortexexe";
+std::filesystem::path GetCortexTemp() {
+    return std::filesystem::temp_directory_path() / "cortex_temp";
+}
+}  // namespace
+
```

Comment:
I think Tests need to cover for variants, i.e. Nightly, Beta, otherwise some intern will break it a year from now

Reply:
> I think Tests need to cover for variants, i.e. Nightly, Beta, otherwise some intern will break it a year from now

I think the function we need to test is `ReplaceBinaryInflight`. It's not related to the variants. 
The download binary part will need to add more logic after James finishes the pause/resume download feature. Will add the unit tests for it later.